src/Admin/BaseMediaAdmin.php

```diff
@@ -161,13 +183,9 @@ protected function configureListFields(ListMapper $list)
 
     protected function configureFormFields(FormMapper $form)
     {
-        $media = $this->getSubject();
-
-        if (!$media) {
-            $media = $this->getNewInstance();
-        }
+        $media = $this->hasSubject() ? $this->getSubject() : $this->getNewInstance();
```

Comment:
I'm not sure it makes a lot of sens since SonataAdmin have a check to throw an error if there is no subject when building the form/show.

Reply:
I agree, but if we throw an exception in this case, it could be considered a BC-break, since it isn't the current behavior.

Reply:
Since there is the exception for 4.0 https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/Admin/AbstractAdmin.php#L896 and a deprecation for 3.x, I would not add the `hasSubject` check here.

And we could add a next major to explain that in 4.0, `$media = $this->getSubject()` will be enough, without the null check.

Reply:
I think I don't get your point, as `$media` is being used several times in this method trusting it is always an instance of `MediaInterface`.

Reply:
In next major
```
$media = $this->hasSubject() ? $this->getSubject() : $this->getNewInstance();
```
can be replaced to
```
$media = $this->getSubject();
```

Reply:
I agree that is the proper behavior for the next major, but I don't understand what are you suggesting here for the 3.x branch, where `$this->getSubject()` returns `null`. Could you please provide a suggestion?

Reply:
I just thought about adding a `Next_major` comment in order to not forget to do the change during the merge into master.

```
// NEXT_MAJOR: use the next line instead.
$media = $this->hasSubject() ? $this->getSubject() : $this->getNewInstance();
// $media = $this->getSubject();
```